packages/application-extension/src/index.ts

```diff
@@ -556,6 +590,114 @@ const topVisibility: JupyterFrontEndPlugin<void> = {
   autoStart: true
 };
 
+/**
+ * Plugin to toggle the left or right sidebar's visibility.
+ */
+const sidebarVisibility: JupyterFrontEndPlugin<void> = {
+  id: '@jupyter-notebook/application-extension:sidebar',
+  requires: [INotebookShell, ITranslator],
+  optional: [IMainMenu, ISettingRegistry, INotebookTools],
+  activate: (
+    app: JupyterFrontEnd<JupyterFrontEnd.IShell>,
+    notebookShell: INotebookShell,
+    translator: ITranslator,
+    menu: IMainMenu | null,
+    settingRegistry: ISettingRegistry | null,
+    notebookTools: INotebookTools | null
+  ) => {
+    if (!sidePanelsEnabled()) {
+      return;
+    }
+
+    const trans = translator.load('notebook');
+
+    /* Arguments for togglePanel command:
+     * side, left or right area
+     * title, widget title to show in the menu
+     * id, widget ID to activate in the sidebar
+     */
+    app.commands.addCommand(CommandIDs.togglePanel, {
+      label: args => args['title'] as string,
+      caption: args => {
+        // We do not substitute the parameter into the string because the parameter is not
+        // localized (e.g., it is always 'left') even though the string is localized.
+        if (args['side'] === 'left') {
+          return trans.__(
+            'Show %1 in the left sidebar',
+            args['title'] as string
+          );
+        } else if (args['side'] === 'right') {
+          return trans.__(
+            'Show %1 in the right sidebar',
+            args['title'] as string
+          );
+        }
+        return trans.__('Show %1 in the sidebar', args['title'] as string);
+      },
+      execute: args => {
+        switch (args['side'] as string) {
+          case 'left':
+            if (notebookShell.leftCollapsed) {
+              notebookShell.activateById(args['id'] as string);
+            } else {
+              notebookShell.collapseLeft();
+              if (notebookShell.currentWidget) {
+                notebookShell.activateById(notebookShell.currentWidget.id);
+              }
+            }
+            break;
+          case 'right':
+            if (notebookShell.rightCollapsed) {
+              notebookShell.activateById(args['id'] as string);
+            } else {
+              notebookShell.collapseRight();
+              if (notebookShell.currentWidget) {
+                notebookShell.activateById(notebookShell.currentWidget.id);
+              }
+            }
+            break;
+        }
+      },
+      isToggled: args => {
+        if (notebookShell.leftCollapsed) {
+          return false;
+        }
+        const currentWidget = notebookShell.leftHandler.current;
+        if (!currentWidget) {
+          return false;
+        }
+
+        return currentWidget.id === (args['id'] as string);
+      }
+    });
+
+    app.restored.then(() => {
+      // Create a menu entry for left and right panel and update it with current widgets
+      if (menu) {
```

Comment:
Maybe these entries could also be added to the command palette? 

We might need to prepend the label with "Show" so it's easier to understand what it's about when the "Table of Contents" entry shows up as a palette entry?

![image](https://user-images.githubusercontent.com/591645/183422735-573b710d-3be6-4917-9643-ea89bd84abeb.png)

Reply:
Fixed in https://github.com/jupyter/notebook/pull/6487/commits/52f10ede26493331a2db77fa08d0ce3e65470190 and https://github.com/jupyter/notebook/pull/6487/commits/c6bc3ef2796f3ae686f038d12398b1989d2e8f4f